### PR TITLE
JWT Expiration Check

### PR DESCRIPTION
**Pull request by darrenwhite24**
## Description
This PR adds an expiration check _before_ the token is verified. For context, we've recently received a lot of Okta signing key errors and if the token has also expired, the expiration check should quiet some of those errors and trigger an attempt to refresh the token.

### CODE REVIEW

**Review by tchamblee — APPROVED** [review]
I don't have too much context with authentic but this looks good to me.

File: index.js [hunk 1]
```diff
@@ -71,6 +71,16 @@ const factory = options => {
     opts.issWhitelist.indexOf(token.payload.iss) > -1 ||
     reject(new IssWhitelistError(`iss '${token.payload.iss}' not in issWhitelist`))
 
+  const checkExp = ({ payload }) => {
+    const { exp } = payload
```
Comment (mgreystone) [inline index.js]: What if `exp` is not present? Does this change effectively require the `exp` field?

File: index.js [hunk 2]
```diff
@@ -71,6 +71,16 @@ const factory = options => {
...
+    const { ignoreExpiration } = verifyOpts
+    const { TokenExpiredError } = jwt
+    const isExpired = !ignoreExpiration && exp < Date.now() / 1000
+    const error = new TokenExpiredError('jwt expired', new Date(exp * 1000))
```
Comment (mgreystone) [inline index.js]: Maybe try not to construct an error if it's possible we won't use it? Some error tracking libs attach to the `Error` constructor.

**Comment by mgreystone** [comment]
Tests would be neat too.

**Comment by darrenwhite24** [comment]
@mgreystone there is already a test for expired tokens. The difference is that it now tests the logic added in this PR rather than the `verify` function from `jsonwebtoken`

**Comment by mgreystone** [comment]
Right. A test for that would be neat. 😉 Ensure `jsonwebtoken.verify` never is called for expired tokens?

**Review by mgreystone — APPROVED** [review]

**Comment by darrenwhite24** [comment]
@mgreystone ahh okay. Yeah, sure.

**Review by PurrBiscuit — APPROVED** [review]

**Comment by jameswelle** [comment]
@darrenwhite24 @mgreystone @tchamblee @PurrBiscuit is this work still planned or has it been abandoned?

**Comment by jameswelle** [comment]
@darrenwhite24 @mgreystone @tchamblee @PurrBiscuit can we get a status update on this?

**Comment by cdwills** [comment]
@jameswelle This is still planned, not abandoned; I can ping you on slack and talk about getting it re-prioritized if it is needed.

**Comment by cdwills** [comment]
Closing in favor of #23

**Pull request closed without merging**